### PR TITLE
adds python testing sections

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

Adds sections to the Testing page for Python.

[Preview](https://temporal-documentation-f1gssdxas.preview.thundergun.io/application-development/testing?lang=python#run-an-activity)

### CODE REVIEW

File: docs/application-development/testing.md [hunk 1]
```diff
@@ -111,7 +113,23 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+To test a Heartbeat in an Activity, you can use the `on_heartbeat()` property of the `ActivityEnvironment` class. This property allows you to set a custom function that will be called every time the `activity.heartbeat()` function is called within the Activity.
+
+```python
+async def test_heartbeats():
+    async def activity_with_heartbeats(param: str):
```
Comment (cretz) [inline docs/application-development/testing.md:120]: May want to move this outside and add a `@activity.defn`. Not required but it can be clearer for docs.

File: docs/application-development/testing.md [hunk 2]
```diff
@@ -168,7 +186,42 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+To cancel an Activity in Temporal, you can use the [`WorkflowHandle.cancel()`](https://python.temporal.io/temporalio.client.WorkflowHandle.html#cancel) method, which will send a cancellation request to the running Workflow.
```
Comment (cretz) [inline docs/application-development/testing.md:189]: DId you mean "To cancel a Workflow"?

File: docs/application-development/testing.md [hunk 3]
```diff
@@ -168,7 +186,42 @@ Content is currently unavailable.
...
+        )
+
+        # Wait for the Activity to start
+        await asyncio.sleep(5)
+
+        # Cancel the workflow
+        await handle.cancel()
+
+        # Assert that the workflow has been canceled
+        assert "CANCELED" == (await handle.describe()).status
```
Comment (cretz) [inline docs/application-development/testing.md:221]: `temporalio.client.WorkflowExecutionStatus.CANCELED` would be better to compare against

File: docs/application-development/testing.md [hunk 4]
```diff
@@ -168,7 +186,42 @@ Content is currently unavailable.
...
+```
+
+The async `cancel()` method sends a cancellation request to the running Workflow. If the Workflow is running an Activity that listens to cancellation, it will terminate the Activity.
```
Comment (cretz) [inline docs/application-development/testing.md:224]:
```suggestion
The async `cancel()` method sends a cancellation request to the running Workflow. If the Workflow is running an, it will cancel the Activity.
```

File: docs/application-development/testing.md [hunk 5]
```diff
@@ -313,7 +366,46 @@ $this->activityMocks->expectFailure('SimpleActivity.echo', new \LogicException('
...
+        # Execute your Workflow as usual
+        assert "Hello, World from mocked activity!" == await client.execute_workflow(
+            GreetingWorkflow.run,
+            "World",
+            id=str(uuid.uuid4()),
+            task_queue=task_queue_name,
+        )
+```
+
+When providing the Worker with the mocked Activity implementations, you should only provide the mocked implementations and not the real implementations.
```
Comment (cretz) [inline docs/application-development/testing.md:406]: I am not sure I understand this sentence, it's a bit of a tautology.

File: docs/application-development/testing.md [hunk 6]
```diff
@@ -583,7 +679,18 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+To implement manual time skipping, use the [`start_time_skipping()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#start_time_skipping) static method.
```
Comment (cretz) [inline docs/application-development/testing.md:682]:
```suggestion
To implement time skipping, use the [`start_time_skipping()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#start_time_skipping) static method.
```

File: docs/application-development/testing.md [hunk 7]
```diff
@@ -583,7 +679,18 @@ Content is currently unavailable.
...
+
+```python
+from temporalio.testing import WorkflowEnvironment
+
+async def test_manual_time_skipping():
+    async with await WorkflowEnvironment.start_time_skipping() as env:
```
Comment (cretz) [inline docs/application-development/testing.md:688]: It is important to note is that time auto-skips workflow tie and then stops auto-skipping during activity execution.

File: docs-src/python/how-to-skip-time-set-up-in-python.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (cretz) [inline docs-src/python/how-to-skip-time-set-up-in-python.md:1]: Maybe not this issue, but we need to explain how the time-skipping test server works in Java, TS, and Python

File: docs-src/python/how-to-skip-time-set-up-in-python.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+title: How to set up time skipping in Python
+sidebar_label: Set up time skipping
+description: Skip time from the test server automatically.
+tags:
+  - python
+  - how-to
+  - time-skipping
+---
+
+To set up a time skipping environment, import [`WorkflowEnvironment`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html) from the [`testing`](https://python.temporal.io/temporalio.testing.html) module.
```
Comment (cretz) [inline docs-src/python/how-to-skip-time-set-up-in-python.md:12]: This doesn't setup an environment, this just does an import. That doesn't really do anything.

File: docs-src/python/how-to-test-cancel-an-activity-in-python.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
+---
+id: how-to-test-cancel-an-activity-in-python
+title: How to test Cancel in an Activity in Python
```
Comment (cretz) [inline docs-src/python/how-to-test-cancel-an-activity-in-python.md:3]: But the contents of the file is about cancelling a workflow

File: docs-src/typescript/clients.md [hunk 11]
```diff
@@ -41,13 +41,15 @@ import { example } from './workflows';
 
 async function run() {
   // Connect to the default Server location (localhost:7233)
+  console.log(Date.now());
```
Comment (cretz) [inline docs-src/typescript/clients.md:44]: Why add this log in the docs?

File: docs-src/typescript/how-to-log-from-a-workflow-in-typescript.md [hunk 12]
```diff
@@ -41,16 +41,18 @@ Explicitly declaring a Sink's interface is optional, but is useful for ensuring
 
 <!--SNIPSTART typescript-logger-sink-interface-->
 
-[packages/test/src/workflows/definitions.ts](https://github.com/temporalio/sdk-typescript/blob/master/packages/test/src/workflows/definitions.ts)
+[sinks/src/workflows.ts](https://github.com/temporalio/samples-typescript/blob/master/sinks/src/workflows.ts)
 
 ```ts
-import type { Sinks } from '@temporalio/workflow';
+import { LoggerSinks, proxySinks, Sinks } from '@temporalio/workflow';
```
Comment (cretz) [inline docs-src/typescript/how-to-log-from-a-workflow-in-typescript.md:47]: Should we change the title/description of the PR to include that there are TS changes too? Even if automatically obtained, this PR does more than add Python testing sections, it also syncs other code.

File: docs/activities.md [hunk 13]
```diff
@@ -13,7 +13,7 @@ import TabItem from '@theme/TabItem';
...
...">Activity Definition<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><div class="tdlpc"><p class="tdlppt">What is an Activity Definition?</p><p class="tdlppd">An Activity Definition is the code that defines the constraints of an Activity Task Execution.</p><p class="tdlplm"><a class="tdlplma" href="#activity-definition">Learn more</a></p></div></a>, an <a class="tdlp" href="#activity-type">Activity Type<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><div class="tdlpc"><p class="tdlppt">What is an Activity Type?</p><p class="tdlppd">An Activity Type is the mapping of a name to an Activity Definition.</p><p class="tdlplm"><a class="tdlplma" href="#activity-type">Learn more</a></p></div></a>, or an <a class="tdlp" href="#activity-execution">Activity Execution<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><div class="tdlpc"><p class="tdlppt">What is an Activity Execution?</p><p class="tdlppd">An Activity Execution is the full chain of Activity Task Executions.</p><p class="tdlplm"><a class="tdlplma" href="#activity-execution">Learn more</a></p></div></a>.
```
Comment (cretz) [inline docs/activities.md:16]: Curious, where did these changes come from? There are a lot of markdown changes here and the next several files that are going to get changed as part of this "Python Testing PR" that are completely unrelated to the PR goals herein.

File: docs-src/typescript/how-to-log-from-a-workflow-in-typescript.md (hunk 12 above)
Comment (cretz) [reply docs-src/typescript/how-to-log-from-a-workflow-in-typescript.md:47]: I suggest making that work in a separate PR

**Review by cretz — APPROVED** [review]

File: docs/application-development/testing.md [hunk 14]
```diff
@@ -954,7 +1020,7 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+One recommended framework for testing in Python for the Temporal SDK is [pytest](https://docs.pytest.org/), which can help with fixtures to stand up and tear down test environments, provide useful test discovery, and make it easy to write parametrized tests.
```
Comment (DJSanti) [inline docs/application-development/testing.md:1023]:
```suggestion
Temporal recommends [pytest](https://docs.pytest.org/) for testing in Python.
This library makes it easy to write parametrized tests, and provides useful test discovery and test environment fixtures.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+sidebar_label: Heartbeat an Activity in a test
+description: To listen to a Heartbeat in a test, use the on_heartbeat property.
+tags:
+  - developer-guide
+  - sdk
+  - python
+  - testing
+---
+
+To test a Heartbeat in an Activity, you can use the `on_heartbeat()` property of the `ActivityEnvironment` class. This property allows you to set a custom function that will be called every time the `activity.heartbeat()` function is called within the Activity.
```
Comment (DJSanti) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:13]:
```suggestion
To test a Heartbeat in an Activity, use the `on_heartbeat()` property of the `ActivityEnvironment` class. This property sets a custom function that will be called every time `activity.heartbeat()` is called within the Activity.
```

File: docs-src/python/how-to-mock-activities-in-python.md [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+        # Execute your Workflow as usual
+        assert "Hello, World from mocked activity!" == await client.execute_workflow(
+            GreetingWorkflow.run,
+            "World",
+            id=str(uuid.uuid4()),
+            task_queue=task_queue_name,
+        )
+```
+
+The mocked Activity implementation should have the same signature as the real implementation, including the input and output types, and the same name. So that when the Workflow invokes the Activity, it will invoke the mocked implementation instead of the real one, allowing you to test your Workflow isolated.
```
Comment (DJSanti) [inline docs-src/python/how-to-mock-activities-in-python.md:49]:
```suggestion
The mocked Activity implementation should have the same signature as the real implementation(input and output types, along with name).
When the Workflow invokes the Activity, it invokes the mocked implementation instead of the real Activity, allowing you to test your Workflow isolated.
```

File: docs-src/python/how-to-run-an-activity-in-python.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+sidebar_label: Run an Activity in a test
+description: To run an Activity in a test for Python, use the `ActivityEnvironment` class.
+tags:
+  - developer-guide
+  - sdk
+  - python
+  - testing
+---
+
+To run an Activity in a test, you can use the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
```
Comment (DJSanti) [inline docs-src/python/how-to-run-an-activity-in-python.md:13]:
```suggestion
To run an Activity in a test, use the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
```

File: docs/application-development/testing.md (hunk 14 above)
Comment (DJSanti) [reply docs/application-development/testing.md:1023]: It's either a class or library. I'm a bit rusty on what Python calls it.

File: docs-src/python/how-to-run-an-activity-in-python.md [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+
+This class allows you to run any callable inside an activity context, and you can use it to test the behavior of your code under different conditions.
```
Comment (DJSanti) [inline docs-src/python/how-to-run-an-activity-in-python.md:15]:
```suggestion
This class allows you to run any callable inside an Activity context. Use it to test the behavior of your code under different conditions.
```

File: docs-src/python/how-to-skip-time-skip-automatically-in-python.md [hunk 19]
```diff
@@ -6,10 +6,11 @@ description: Skip time from the test server automatically.
 tags:
   - python
   - how-to
+  - time-skipping
 ---
 
 Use the [`start_time_skipping()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#start_time_skipping) method to start a test server process and skip time automatically.
 
 Use the [`start_local()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#start_local) method for a full local Temporal server.
```
Comment (DJSanti) [inline docs-src/python/how-to-skip-time-skip-automatically-in-python.md:14]: To launch the server vs. going to the existing client one?
This should get a bit more elaboration.

File: docs-src/python/how-to-use-test-frameworks-in-python.md [hunk 20]
```diff
@@ -0,0 +1,12 @@
...
+title: How to use test frameworks in Python
+sidebar_label: Test frameworks
+description: Use Pytest for testing frameworks in Python.
+tags:
+  - developer-guide
+  - python
+  - testing
+---
+
+One recommended framework for testing in Python for the Temporal SDK is [pytest](https://docs.pytest.org/), which can help with fixtures to stand up and tear down test environments, provide useful test discovery, and make it easy to write parametrized tests.
```
Comment (DJSanti) [inline docs-src/python/how-to-use-test-frameworks-in-python.md:12]:
```suggestion
Temporal recommend [pytest](https://docs.pytest.org/) for testing code in the Python SDK. This library provides fixtures to stand up and tear down test environments, along with useful test discovery and methods to write parametrized tests.
```

Comment (DJSanti) [reply docs-src/python/how-to-use-test-frameworks-in-python.md:12]: I know I wrote something similar down below. Again, if it's library or class, please fix. Or ignore it entirely. Up to you.

File: docs/application-development/testing.md [hunk 21]
```diff
@@ -54,7 +54,9 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+To run an Activity in a test, you can use the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
```
Comment (DJSanti) [inline docs/application-development/testing.md:57]:
```suggestion
To run an Activity in a test, use the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
```

File: docs/application-development/testing.md [hunk 22]
```diff
@@ -54,7 +54,9 @@ Content is currently unavailable.
...
+
+This class allows you to run any callable inside an activity context, and you can use it to test the behavior of your code under different conditions.
```
Comment (DJSanti) [inline docs/application-development/testing.md:59]:
```suggestion
This class allows you to run any callable inside an Activity context. Use it to test the behavior of your code under different conditions.
```

Comment (DJSanti) [reply docs/application-development/testing.md:59]: Also, callable what?

File: docs/application-development/testing.md [hunk 23]
```diff
@@ -111,7 +113,23 @@ Content is currently unavailable.
 </TabItem>
 <TabItem value="python">
 
-Content is currently unavailable.
+To test a Heartbeat in an Activity, you can use the `on_heartbeat()` property of the `ActivityEnvironment` class. This property allows you to set a custom function that will be called every time the `activity.heartbeat()` function is called within the Activity.
```
Comment (DJSanti) [inline docs/application-development/testing.md:116]:
```suggestion
To test a Heartbeat in an Activity, use the `on_heartbeat()` property in the `ActivityEnvironment` class to set a custom function. This function will be called every time the `activity.heartbeat()` function is called within the Activity.
```

Comment (DJSanti) [reply docs/application-development/testing.md:116]: Just realized this was the generated doc. Please apply these changes to the source docs if desired.

**Review by DJSanti — CHANGES_REQUESTED** [review]
Make sure Chad approves of the more detailed Python stuff. I can check again after that.

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+sidebar_label: Heartbeat an Activity in a test
+description: To listen to a Heartbeat in a test, use the on_heartbeat property.
+tags:
+  - developer-guide
+  - sdk
+  - python
+  - testing
+---
+
+To test a Heartbeat in an Activity, use the [`on_heartbeat()`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html#on_heartbeat) property of the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class. This property sets a custom function that will be called every time the `activity.heartbeat()` function is called within the Activity.
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:13]:
```suggestion
To test a Heartbeat in an Activity, use the [`on_heartbeat()`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html#on_heartbeat) property of the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
This property sets a custom function that is called every time the `activity.heartbeat()` function is called within the Activity.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+```python
+@activity.defn
+async def activity_with_heartbeats(param: str):
+    activity.heartbeat(f"param: {param}")
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:18]:
```suggestion
    activity.heartbeat("param: {param}")
```
Is the `f` in the argument some weird Python casting? A stray character? Missing an `=`? Inquiring minds 'nd whatnot...

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+    activity.heartbeat("second heartbeat")
+
+env = ActivityEnvironment()
+heartbeats = []
+# Set the `on_heartbeat` property to a callback function that will be called for each heartbeat sent by the activity.
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:23]:
```suggestion
# Set the `on_heartbeat` property to a callback function that will be called for each Heartbeat sent by the Activity.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+env.on_heartbeat = lambda *args: heartbeats.append(args[0])
+# Use the run method to start the activity, passing in the function that contains the heartbeats and any necessary parameters.
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:25]:
```suggestion
# Use the run method to start the activity, passing in the function that contains the Heartbeats and any necessary parameters.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+await env.run(activity_with_heartbeats, "test")
+# Verify that the expected heartbeats are received by the callback function.
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:27]:
```suggestion
# Verify that the expected Heartbeats are received by the callback function.
```

File: docs-src/python/how-to-mock-activities-in-python.md [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+Provide mock Activity implementations to the Worker.
+
+```python
+import uuid
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+# Import your Activity definition and real implementation
```
Comment (djmagee) [inline docs-src/python/how-to-mock-activities-in-python.md:19]:
```suggestion
# Import your Activity Definition and real implementation
```

File: docs-src/python/how-to-run-an-activity-in-python.md [hunk 30]
```diff
@@ -0,0 +1,15 @@
...
+tags:
+  - developer-guide
+  - sdk
+  - python
+  - testing
+---
+
+To run an Activity in a test, use the [`ActivityEnvironment`](https://python.temporal.io/temporalio.testing.ActivityEnvironment.html) class.
+
+This class allows you to run any callable inside an Activity context. Use it to test the behavior of your code under different conditions.
```
Comment (djmagee) [inline docs-src/python/how-to-run-an-activity-in-python.md:15]:
```suggestion
This class allows you to run any callable inside an Activity context.
Use it to test the behavior of your code under various conditions.
```
Per https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/d/different.

File: docs-src/python/how-to-skip-time-skip-automatically-in-python.md [hunk 31]
```diff
@@ -6,10 +6,11 @@ description: Skip time from the test server automatically.
...
 
-Use the [`from_client`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#from_client) method for an existing Temporal sever.
+Use the [`from_client()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#from_client) method for an existing Temporal sever.
```
Comment (djmagee) [inline docs-src/python/how-to-skip-time-skip-automatically-in-python.md:16]:
```suggestion
Use the [`from_client()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#from_client) method for an existing Temporal Server.
```

File: docs-src/python/how-to-skip-time-skip-manually-in-python.md [hunk 32]
```diff
@@ -0,0 +1,23 @@
+---
+id: how-to-skip-time-skip-manually-in-python
+title: How to skip time manually in Python
+sidebar_label: Skip time manually
+description: Skip time from the test server manually.
```
Comment (djmagee) [inline docs-src/python/how-to-skip-time-skip-manually-in-python.md:5]:
```suggestion
description: To implement time skipping, use the `start_time_skipping()` static method.

File: docs-src/python/how-to-use-test-frameworks-in-python.md (hunk 20 above)
Comment (djmagee) [inline docs-src/python/how-to-use-test-frameworks-in-python.md:12]:
```suggestion
One recommended framework for testing in Python for the Temporal SDK is [pytest](https://docs.pytest.org/), which can help with fixtures to stand up and tear down test environments, provide useful test discovery, and make it easy to write parameterized tests.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md (hunk 25 above)
Comment (djmagee) [reply docs-src/python/how-to-listen-to-heartbeats-in-python.md:19]: TIL something. Been a while since I coded in Python, obvs.

File: docs-src/python/how-to-mock-activities-in-python.md [hunk 33]
```diff
@@ -0,0 +1,49 @@
...
+        # Execute your Workflow as usual
+        assert "Hello, World from mocked activity!" == await client.execute_workflow(
+            GreetingWorkflow.run,
+            "World",
+            id=str(uuid.uuid4()),
+            task_queue=task_queue_name,
+        )
+```
+
+The mocked Activity implementation should have the same signature as the real implementation, including the input and output types, and the same name. So that when the Workflow invokes the Activity, it will invoke the mocked implementation instead of the real one, allowing you to test your Workflow isolated.
```
Comment (djmagee) [inline docs-src/python/how-to-mock-activities-in-python.md:49]:
```suggestion
The mocked Activity implementation should have the same signature as the real implementation (including the input and output types) and the same name.
When the Workflow invokes the Activity, it invokes the mocked implementation instead of the real one, allowing you to test your Workflow isolated.
```

File: docs-src/python/how-to-listen-to-heartbeats-in-python.md [hunk 34]
```diff
@@ -0,0 +1,30 @@
...
+@activity.defn
+async def activity_with_heartbeats(param: str):
+    activity.heartbeat(f"param: {param}")
+    activity.heartbeat("second heartbeat")
+
+env = ActivityEnvironment()
+heartbeats = []
+# Set the `on_heartbeat` property to a callback function that will be called for each Heartbeat sent by the Activity.
+env.on_heartbeat = lambda *args: heartbeats.append(args[0])
+# Use the run method to start the activity, passing in the function that contains the Heartbeats and any necessary parameters.
```
Comment (djmagee) [inline docs-src/python/how-to-listen-to-heartbeats-in-python.md:26]:
```suggestion
# Use the run method to start the Activity, passing in the function that contains the Heartbeats and any necessary parameters.
```
Oops. I missed this one during my previous review. Mea culpa. ("I done boo-booed.")

File: docs-src/python/how-to-skip-time-skip-automatically-in-python.md (hunk 19 above)
Comment (djmagee) [inline docs-src/python/how-to-skip-time-skip-automatically-in-python.md:14]:
```suggestion
Use the [`start_local()`](https://python.temporal.io/temporalio.testing.WorkflowEnvironment.html#start_local) method for a full local Temporal Server.
```

**Review by djmagee — COMMENTED** [review]
Three more tweaks whilst you await Samantha's review.

**Review by DJSanti — APPROVED** [review]

**Review by djmagee — APPROVED** [review]
